src/components/homepage/HeroSection/HeroHighlightSchema.jsx

```diff
@@ -0,0 +1,52 @@
+import * as Yup from "yup"
+
+import {
+  HERO_BUTTON_MIN_LENGTH,
+  HERO_BUTTON_MAX_LENGTH,
+  HIGHLIGHTS_DESCRIPTION_MIN_LENGTH,
+  HIGHLIGHTS_DESCRIPTION_MAX_LENGTH,
+  HIGHLIGHTS_TITLE_MIN_LENGTH,
+  HIGHLIGHTS_TITLE_MAX_LENGTH,
+} from "utils/validators"
+
+export const HeroHighlightOptionSchema = Yup.object({
+  title: Yup.string()
+    .min(
+      HIGHLIGHTS_TITLE_MIN_LENGTH,
+      `Title must be longer than ${HIGHLIGHTS_TITLE_MIN_LENGTH} characters`
+    )
+    .max(
+      HIGHLIGHTS_TITLE_MAX_LENGTH,
+      `Title must be shorter than ${HIGHLIGHTS_TITLE_MAX_LENGTH} characters`
+    ),
+  url: Yup.string(),
+  description: Yup.string()
+    .min(
+      HIGHLIGHTS_DESCRIPTION_MIN_LENGTH,
+      `Title must be longer than ${HIGHLIGHTS_DESCRIPTION_MIN_LENGTH} characters`
+    )
+    .max(
+      HIGHLIGHTS_DESCRIPTION_MAX_LENGTH,
+      `Title must be shorter than ${HIGHLIGHTS_DESCRIPTION_MAX_LENGTH} characters`
+    ),
+})
+
+export const HeroHighlightSchema = Yup.object().shape({
+  button: Yup.lazy((val) =>
+    val
+      ? Yup.string()
+          .min(
+            HERO_BUTTON_MIN_LENGTH,
+            `Title must be longer than ${HERO_BUTTON_MIN_LENGTH} characters`
+          )
+          .max(
+            HERO_BUTTON_MAX_LENGTH,
+            `Title must be shorter than ${HERO_BUTTON_MAX_LENGTH} characters`
+          )
+      : Yup.string().nullable()
+  ),
+  url: Yup.lazy((val) => (val ? Yup.string() : Yup.string().nullable())),
+  key_highlights: Yup.lazy((val) =>
+    val ? Yup.array().of(HeroHighlightOptionSchema) : Yup.object().nullable()
+  ),
```

Comment:
just checking - why would chaining `.nullable` not work? my understanding of it is that if it's empty (ie, `null`), then it passes but if it isn't, then it'd be ran through the validator.

Reply:
372d3068